File: src/backend/Seems.Domain/Entities/Page.cs
using Seems.Domain.Common;
using Seems.Domain.Enums;
using Seems.Domain.ValueObjects;

namespace Seems.Domain.Entities;

public class Page : BaseEntity, IAuditable
{
    public Guid? ParentId { get; set; }
    public string Slug { get; set; } = string.Empty;
    public string Path { get; set; } = string.Empty;
    public int SortOrder { get; set; }
    public string Title { get; set; } = string.Empty;
    public string TemplateKey { get; set; } = string.Empty;
    public string? ThemeKey { get; set; }
    public SeoMeta Seo { get; set; } = new();
    public ContentStatus Status { get; set; } = ContentStatus.Draft;
    public bool IsDefault { get; set; }
    public bool ShowInNavigation { get; set; } = true;

    public Page? Parent { get; set; }
    public ICollection<Page> Children { get; set; } = [];
    public ICollection<SlotMapping> Slots { get; set; } = [];
}

File: src/frontend/public/types/page.ts
export interface SeoMeta {
  title: string
  description?: string
  ogTitle?: string
  ogDescription?: string
  ogImage?: string
  canonical?: string
  robots?: string
}

export interface SlotMapping {
  slotKey: string
  targetType: 'Content' | 'Module'
  targetId: string
  order: number
  parameters?: Record<string, unknown> | null
}

export interface Page {
  id: string
  slug: string
  title: string
  templateKey: string
  themeKey?: string
  seo: SeoMeta
  slots: SlotMapping[]
  status: 'draft' | 'published'
  createdAt: string
  updatedAt: string
  /** Populated when the page was matched via a parametric path pattern (e.g. blog/:id). */
  urlParameters?: Record<string, string> | null
}
